### ISSUE TITLE
Traefik IngressRoute object lacks type checking

### ISSUE THREAD

**Issue by trondhindenes** [containous/traefik#5740]
The "port" field in an IngressRoute object (`v1alpha1.Route.Services`) lacks type definitions. This means that it's possible to create an invalid IngressRoute object, for instance 
```yaml
spec:
  entryPoints:
  - http
  routes:
  - kind: Rule
    match: Host(`test.com`) && PathPrefix(`/`)
    services:
    - name: test
      port: http
```
By leveraging type definitions in CRDs (`type: integer`)  this could be validated at deploy-time, so that we never create invalid objects.

**Comment by dtomcej** [comment]
We should probably support named ports in the CRD specs.

**Comment by trondhindenes** [comment]
you mean support both? In that case I would do 
```
    services:
    - name: test
      port: 80
      portName: http
```
And then made `port` and `portName` mutually exclusive, again with type checking for the `port` field since that would always be `int`. Make it as hard as possible to make mistakes.

**Comment by ldez** [comment]
In the Ingress object (not IngressRoute) the port has a double type string/int, so we think we can do the same with IngressRoute.

**Comment by trondhindenes** [comment]
That makes sense.

**Comment by ldez** [comment]
Closed in favor of #5740
